### ISSUE TITLE
[FEATURE] Better handling of IRs in Amp IRs 

### ISSUE THREAD

**Issue by Boone777** [Chowdhury-DSP/BYOD#246]
**Is your feature request related to a problem? Please describe.**
When loading custom IRs, going through a bunch of them is tedious since I have to pop open a window and go through a bunch of folders to find my cab IR. Rinse and repeat 50-100 times.

**Describe the solution you'd like**
- Ability to set default folder
- Ability to scroll up and down IRs with mouse wheel over the box 
- And/Or arrows next to the loaded Ir box: previous and next
-A stereo IR cab loader ( Or quad) with ability to blend between IRs (X/Y blend) would tidy up things instead of routing multiple IR loaders.

As exemples the freebie Nadir from Ignite amp or Libra from Ignite.

**Comment by jatinchowdhury18** [comment]
Thanks for the feature request! I think generally these are some good ideas, but overall, this maybe something that needs a bit more thought.

In most cases, I actually don't use the Amp IRs module in BYOD, and instead use Libra (or something similar) in my signal chain after BYOD. There's a few reasons for this:
1. I usually would use Amp IRs as the final module in my signal chain, so there's no difference in the sound I'm getting.
2. As you mentioned, dedicated IR-loading/cab sim plugins usually have more features, and ease-of-use improvements.
3. Since BYOD is usually used with oversampling, the IR processing usually has to happen at an unnecessarily high sample rate, which wastes a lot of CPU depending on how much oversampling is being used.

To elaborate on this last point, most guitar cab IRs filter out most of the signal above 8-10 kHz, so the IR processing can be done at 44.1 kHz samples rate (or sometimes even lower), without losing any of the audible features of the IR frequency response. Most DAW users are probably running at either 44.1 or 48 kHz, so with BYOD at the default 2x oversampling, the IR is being processed at 88.2 or 96 kHz, which wastes some CPU. As you can imagine, going to higher sample rates would waste even more CPU.

With that in mind, maybe a better thing to do might be to focus on making another "section" of BYOD, which would be focused specifically on IR-loading and cabinet simulation. This section would run _after_ the "pedalboard" part of BYOD, and would run without oversampling (or maybe even with undersampling?). Maybe this sort of thing would make more sense as its own dedicated plugin? I'd be curious to hear your thoughts on this idea.

---

Anyway, to address your suggestions more directly, I think having a default folder option where BYOD can look for IRs shouldn't be too hard, so we should definitely do that. Having next/previous arrow buttons also shouldn't be too hard, though I'll have to think about the UI for that, since the IR menu already seems a little bit "cramped" in the horizontal direction.

Being able to "scroll" through the IRs with the mouse wheel might be a bit tricker to implement... I'd have to investigate that a little more.

Stereo or Quad blending might also need a bit more thought. I feel like if we do a dedicated IRs section, then we wouldn't need to have options like that in the Amp IRs module. I guess my main worry is that having those options would encourage folks to use the module in a very CPU-inefficient way, so I'd rather try to work on a way to provide that same functionality but in a more efficient context.

Anyway, those are some of the things I've been thinking about, but I'd be curious to hear your ideas as well!

**Comment by Boone777** [comment]
I really like your idea of having an 'external' cab IR loader. It can be both for CPU efficiency and also can be great to be used on it's own with other amp sims. Thing is AFAIK, there is no decent native cab plugin for Linux. And that would definitly be a most welcome addition to the plugin world of Linux. Meanwhile, just the little extras into the current Ir loader would help immensly.

**Comment by jatinchowdhury18** [comment]
HI there! Sorry it's taken me a while to get back to this.

With #274 we added the ability for users to set a default IR folder, so when they open the plugin all their IRs in that directory will be available and sorted by sub-folder.

With #287 we added the ability to iterate over all the available IRs using either the arrow keys or scroll-wheel. I think the scroll-wheel might be a little too sensitive at the moment, so we might need to tune that interaction a little differently.

Anyway, future [nightly builds](https://chowdsp.com/nightly.html#byod) will have both of these features available, if you'd like to try them out.

I think that's all we're going to do for the version 1.2.0 release. I'm still thinking about the other functionality we had discussed (blending/panning/etc), but I don't yet have a clear idea of how that should fit into BYOD (or maybe it should just be it's own plugin, IDK).
